packages/devtools_app/lib/src/screens/vm_developer/object_inspector/vm_instance_display.dart

```diff
@@ -0,0 +1,244 @@
+  late Future<void> _initialized;
+  late DartObjectNode _root;
+
+  @override
+  void initState() {
+    super.initState();
+    _populate();
+  }
+
+  @override
+  void didUpdateWidget(VmInstanceDisplay oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.instance == oldWidget.instance) {
+      return;
+    }
+    _populate();
+  }
+
+  void _populate() async {
+    final isolateRef = serviceManager.isolateManager.selectedIsolate.value;
+    _root = DartObjectNode.fromValue(
+      name: 'value',
+      value: widget.instance.obj,
+      isolateRef: isolateRef,
+      artificialName: true,
+    );
+
+    _initialized = buildVariablesTree(_root)
+        .then(
+          (_) => _root.expand(),
+        )
+        .then(
+          (_) => Future.wait([
+            for (final child in _root.children) buildVariablesTree(child),
+          ]),
+        );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Split(
+      axis: Axis.vertical,
+      initialFractions: const [0.5, 0.5],
+      children: [
+        OutlineDecoration.onlyBottom(
+          child: _InstanceViewer(
+            controller: widget.controller,
+            instance: widget.instance,
+          ),
+        ),
+        OutlineDecoration.onlyTop(
+          child: Column(
+            children: [
+              const AreaPaneHeader(
+                title: Text('Properties'),
+                needsTopBorder: false,
+              ),
+              Flexible(
+                child: FutureBuilder(
+                  future: _initialized,
+                  builder: (context, snapshot) {
+                    if (snapshot.connectionState != ConnectionState.done)
+                      return const CenteredCircularProgressIndicator();
+                    return ExpandableVariable(
+                      variable: _root,
+                      dataDisplayProvider: (variable, onPressed) {
+                        return DisplayProvider(
+                          controller: widget.controller,
+                          variable: variable,
+                          onTap: onPressed,
+                        );
+                      },
+                    );
+                  },
+                ),
+              ),
+            ],
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _InstanceViewer extends StatelessWidget {
+  const _InstanceViewer({
+    required this.controller,
+    required this.instance,
+  });
+
+  final ObjectInspectorViewController controller;
+  final InstanceObject instance;
+
+  @override
+  Widget build(BuildContext context) {
+    return VmObjectDisplayBasicLayout(
+      controller: controller,
+      object: instance,
+      generalDataRows: [
+        serviceObjectLinkBuilderMapEntry<ClassRef>(
+          controller: controller,
+          key: 'Object Class',
+          object: instance.obj.classRef!,
+        ),
+        selectableTextBuilderMapEntry(
+          'Shallow Size',
```

Comment:
put these "X Size" strings into shared consts somewhere. These are used in vm_developer_common_widgets.dart too